phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/WriteWorkload.java

```diff
@@ -0,0 +1,403 @@
+     * @param parser      {@link org.apache.phoenix.pherf.configuration.XMLConfigParser}
+     * @param scenario    {@link org.apache.phoenix.pherf.configuration.Scenario} If null is passed
+     *                    it will run against all scenarios in the parsers list.
+     * @throws Exception
+     */
+    public WriteWorkload(PhoenixUtil phoenixUtil, Properties properties, XMLConfigParser parser,
+            Scenario scenario) throws Exception {
+        this.pUtil = phoenixUtil;
+        this.parser = parser;
+        this.rulesApplier = new RulesApplier(parser);
+        this.resultUtil = new ResultUtil();
+
+        // Overwrite defaults properties with those given in the configuration. This indicates the
+        // scenario is a R/W mixed workload.
+        if (scenario != null) {
+            this.scenario = scenario;
+            writeParams = scenario.getWriteParams();
+            threadSleepDuration = writeParams.getThreadSleepDuration();
+        } else {
+            writeParams = null;
+            this.scenario = null;
+            threadSleepDuration = 0;
+        }
+
+        int size = Integer.parseInt(properties.getProperty("pherf.default.dataloader.threadpool"));
+
+        this.threadPoolSize = (size == 0) ? Runtime.getRuntime().availableProcessors() : size;
+
+        // TODO Move pool management up to WorkloadExecutor
+        this.pool = Executors.newFixedThreadPool(this.threadPoolSize);
+
+        String
+                bSize =
+                (writeParams == null) || (writeParams.getBatchSize() == Long.MIN_VALUE) ?
+                        properties.getProperty("pherf.default.dataloader.batchsize") :
+                        String.valueOf(writeParams.getBatchSize());
+        this.batchSize =
+                (bSize == null) ? PherfConstants.DEFAULT_BATCH_SIZE : Integer.parseInt(bSize);
+    }
+
+    @Override public void complete() {
+    }
+
+    public Runnable execute() throws Exception {
+        return new Runnable() {
+            @Override public void run() {
+                try {
+                    DataLoadTimeSummary dataLoadTimeSummary = new DataLoadTimeSummary();
+                    DataLoadThreadTime dataLoadThreadTime = new DataLoadThreadTime();
+
+                    if (WriteWorkload.this.scenario == null) {
+                        for (Scenario scenario : getParser().getScenarios()) {
+                            exec(dataLoadTimeSummary, dataLoadThreadTime, scenario);
+                        }
+                    } else {
+                        exec(dataLoadTimeSummary, dataLoadThreadTime, WriteWorkload.this.scenario);
+                    }
+                    resultUtil.write(dataLoadTimeSummary);
+                    resultUtil.write(dataLoadThreadTime);
+
+                } catch (Exception e) {
+                    logger.warn("", e);
+                }
+            }
+        };
+    }
+
+    private synchronized void exec(DataLoadTimeSummary dataLoadTimeSummary,
```

Comment:
No synchronization needed here since no state is being guarded unless you want to prevent more than one thread from executing exec which I am not sure you really need.